os/include/tinyara/watchdog.h

```diff
@@ -273,6 +273,8 @@ void watchdog_unregister(FAR void *handle);
  * Architecture-specific Application Interfaces
  ****************************************************************************/
 
+void up_watchdog_disable(void);
```

Comment:
all functions here is for upperhalf(like a register, unregister), you have to find another way.
Lower should be controlled by only upper. 
In other words, ioctl, I mean

Reply:
It's for internal usages of kernel side and it is required by scheduler or assert regardless of user. Currently it is used in assert only. In assert, we can't use lower/upperhalf structure.

Reply:
I don't agree, there are serveral reasons why do not allow this code.
Also I don't mean that you have to use lower/upperhalf, you have to use file operation.(ioctl I mean) Please don't access lower driver's function directly. see below.

1. each of module.h's definition including upper's interface such as register, unregister.
   Also define lower half's operation, such as watchdog_ops_s.
   All real device should inherits this operation only. So virtual driver(upper) doesn't know addtional function except these function set.
  So, that is, all device don't have to implement function what junchan has added, because it is not mandatory function, just breaked rule.

2. virtual driver allow to access exclusively, so please assume that it is under controlling by the another place normally.
  (for example, there's counting driver, and app is counting number from driver. but system reset counting number before current working is not finished. if count is initialized as a 0 and app trying to mod with 0, then??)

Reply:
What you say is for normal status. In assert status, we have not to use file operation.

Reply:
Will explain you in private channel